code/game/turfs/turf.dm

```diff
@@ -244,7 +244,7 @@
 		for(var/obj/O in contents) //this is for deleting things like wires contained in the turf
 			if(O.level != 1)
 				continue
-			if(O.invisibility == 101)
+			if(O.invisibility == INVISIBILITY_MAXIMUM)
 				O.singularity_act()
```

Comment:
this might fix singularity eating abstract things it shouldn't, but these objects probably already have singularity_act() overriden anyway.